### PR TITLE
use mbc-common for DB initialization

### PR DESCRIPTION
**Pull request by Lacrymology**
depends on #28

merge that first

### CODE REVIEW

**Comment by xaiki** [comment]
this depends on #28

On 5 April 2013 15:07, Tomas Neme notifications@github.com wrote:

> depends on #28
> 
> ---
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/inaes-tic/mbc-mosto Lacrymology/mbc-common
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/inaes-tic/mbc-mosto/pull/29
> 
> Commit Summary
> 
> add mongo-driver file
> add backups to gitignore
> driver skeleton
> basic requirements and simple methods
> TEMPORARY db setup
> load db collections
> read playlists from DB function prototype
> search scheds that overlap with the timeframe now-config.load_time
> find the playlist referenced by the sched in the database, and turn it into
> a Playlist object
> create the Media and Playlist objects, and call the defined callback
> don't need to pass the list as a parameter
> subscribe to mubsub channel and update playlists
> add new requirements
> add the new driver
> use the mongo driver in mosto
> use occurrence id as playlist name (unique)
> make sure _id is an ObjectId before changing it to hex string
> separate C[R]UD message callbacks
> move playlist driver choice to config file
> Merge remote-tracking branch 'origin/master' into
> Lacrymology/playlists/mongo-driver
> add the option to pass config JSON objects to drivers
> Merge branch 'master' into Lacrymology/playlists/mongo-driver
> add mbc-common requirement
> use mbc-common.db instead of our local script
> remove local db init script
> 
> File Changes
> 
> M .gitignore (1)
> M config.json (2)
> A drivers/playlists/conf/mongo-driver.json (3)
> A drivers/playlists/mongo-driver.js (100)
> M drivers/playlists/playlists-driver.js (13)
> M mosto.js (5)
> M package.json (6)
> 
> Patch Links:
> 
> https://github.com/inaes-tic/mbc-mosto/pull/29.patch
> https://github.com/inaes-tic/mbc-mosto/pull/29.diff

## 

Niv Sardi

File: drivers/playlists/mongo-driver.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+var config   = require("./conf/mongo-driver"),
+    Playlist = require('../../api/Playlist'),
+    Media    = require('../../api/Media'),
+    mubsub   = require("mubsub"),
+    moment   = require("moment"),
+    mbc = require('mbc-common');
```
Comment (jmrunge) [inline drivers/playlists/mongo-driver.js]: Indent? :)

File: package.json [hunk 2]
```diff
@@ -23,7 +23,11 @@
         "underscore"  : "1.2.x",
         "node-uuid"   : "1.3.x",
         "xmlbuilder"  : "0.3.x",
-        "watchr"      : ">=2.3"
+        "watchr"      : ">=2.3",
+        "mubsub"      : "git://github.com/inaes-tic/mubsub.git",
```
Comment (jmrunge) [inline package.json:27]: Is this the same as using a submodule?  Differences?

**Comment by jmrunge** [comment]
Ok, reviewed both.  Please Tom look at my comments on both, fix/respond and
re pull.

Thanks!

2013/4/5 Niv Sardi notifications@github.com

> this depends on #28
>

File: package.json (hunk 2 above)
Comment (Lacrymology) [inline package.json:27]: watcha talking about? I don't understand the question

Comment (jmrunge) [inline package.json:27]: I always used git submodules for cloning another repo inside my repo.  Didnt knew you could specify this in package.json.  That was the question. What is the diference between a submodule and this?

Enviado desde mi BlackBerry

-----Original Message-----

Comment (Lacrymology) [inline package.json:27]: ah, the difference is that this way is not git-tracked. It's updated by `npm upgrade` (if we take care to bump the version on the other side) If we DON'T bump the version, then we need to manually delete it from node_modules and `npm install` again

Comment (jmrunge) [inline package.json:27]: I like it, its more clear. We should try to put node-mlt this way too. @csrocha?

Enviado desde mi BlackBerry

-----Original Message-----

**Pull request merged**